k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java

```diff
@@ -1489,12 +1533,28 @@ public void onMessageBuildSuccess(MimeMessage message, boolean isDraft) {
             }
         } else {
             currentMessageBuilder = null;
+
+            if (mAction == Action.REDIRECT) {
+                LocalMessage referenceMessage = mMessageReference.restoreToLocalMessage(getApplicationContext());
+                message = referenceMessage.cloneAsSuper(true);
+                message.setResentDate(new Date(), K9.hideTimeZone());
+                message.setRecipients(RecipientType.RESENT_TO, toArray(recipientPresenter.getToAddresses()));
```

Comment:
thats what the whole thing is about.
of cause k-9 only has to support the client part of adding the headers.